### PR TITLE
fix(cid): localhost should only be considered as proxy if prefix is c or v.

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by cramforce** [comment]
Do the unit tests still go through the same code path?

**Comment by erwinmombay** [comment]
@cramforce with this current code, i can't 100% guarantee that. i think there could certainly be an indirect use of the `cid` service in the tests that used to go through the `getBaseCid` path but would now go through the `Promise.resolve` path. (any test that had cid usage where url is `localhost` w/o `v or c prefix` and used to go through `getBaseCid` but would now go through the `Promise.resolve` path).

im thinking of just removing the changes i made and doing this:

```js

  if (url.origin.indexOf('http://localhost:') != 0) {
    assert(prefix == 'c' || prefix == 'v',
        'Unknown path prefix in url %s', url.href);
  }
```

in `getSourceOrigin` which fixes my localhost warning problems and would ensure all tests still go through the same code path.

**Comment by cramforce** [comment]
I'm confused by the changeset that GH is showing now. Otherwise it was fine.

**Comment by jridgewell** [comment]
This LGTM. What's the `v` prefix for?

**Comment by erwinmombay** [comment]
@cramforce i dont think i was confident that the previous code wouldn't be forcing the tests into the `Promise.resolve` path (a test using localhost:8000/test for example). had to change from `localDev` since `localDev` also checks that the files come from `dist` or `base` (should we just remove that check in localDev?).

not sure what the v/c stands for, @cramforce can you shed some light on that one?

File: src/service/cid-impl.js [hunk 1]
```diff
@@ -147,10 +146,11 @@ export function getSourceOrigin(url) {
   // The /s/ is optional and signals a secure origin.
   const path = url.pathname.split('/');
   const prefix = path[1];
-  const mode = getMode();
-  // whitelist while localdev and file is in build/ or examples/
-  if (!(mode.localDev &&
-        (prefix == 'examples.build' || prefix == 'examples'))) {
+  // whitelist localhost
+  const localDev = location.hostname == 'localhost' ||
```
Comment (cramforce) [inline src/service/cid-impl.js]: I think what we discussed was to instead make `#isProxyOrigin` only return true if the assert would not fail.

Comment (erwinmombay) [inline src/service/cid-impl.js]: @cramforce reverted to original change. PTAL

File: src/service/cid-impl.js [hunk 2]
```diff
@@ -126,10 +125,13 @@ function getExternalCid(cid, externalCidScope, persistenceConsent) {
...
  */
 export function isProxyOrigin(url) {
+  const path = url.pathname.split('/');
+  const prefix = path[1];
   // List of well known proxy hosts. New proxies must be added here
   // to generate correct tokens.
   return (url.origin == 'https://cdn.ampproject.org' ||
-      url.origin.indexOf('http://localhost:') == 0);
+      (url.origin.indexOf('http://localhost:') == 0 &&
+       (prefix == 'c' || prefix == 'v')));
```
Comment (erwinmombay) [inline src/service/cid-impl.js]: @cramforce PTAL. changed to check if localhost has prefix c/v

**Comment by cramforce** [comment]
LGTM

**Pull request merged**